finalitygadget/finalitygadget.go

```diff
@@ -384,8 +424,14 @@ func (fg *FinalityGadget) QueryLatestFinalizedBlock() (*types.Block, error) {
 	return fg.db.QueryLatestFinalizedBlock()
 }
 
-// This function process blocks indefinitely, starting from the last finalized block.
-func (fg *FinalityGadget) ProcessBlocks(ctx context.Context) error {
+// This function is run once at startup and starts the FG from the last finalized block.
+// Note that this logic will fail if the FG is started before the chain has ETH finalized its first block.
+// The intended startup order for new chains is:
+//  1. Integrate FG with it disabled on CW contract
```

Comment:
after my new PR in the OP repo, the OP chain will always be up first. so the order should be:

1. Start the OP chain with `babylonFinalityGadgetRpc` rollup.config set to empty string. wait for it to have at least 1 eth-finalized block
2. ...
3. restart `op-node` after setting `babylonFinalityGadgetRpc` with the FG gRPC url
4. enable finality gadget in the cw contract